consensus/src/consensus_observer/observer.rs

```diff
@@ -277,8 +264,25 @@ impl ConsensusObserver {
             // Remove the committed blocks from the pending blocks
             remove_pending_blocks(pending_blocks, &ledger_info);
 
-            // Update the root ledger info
-            *root.lock() = ledger_info;
+            // Verify the ledger info is for the same epoch
+            let mut root = root.lock();
+            if ledger_info.commit_info().epoch() != root.commit_info().epoch() {
+                error!(
```

Comment:
this is probably warn instead of error since it's just a race condition?

Reply:
Gotcha -- will downgrade 😄